simplesat/constraints/constraint_modifiers.py

```diff
@@ -96,6 +96,21 @@ def update(self, other_modifiers):
         self.allow_newer.update(other_modifiers.allow_newer)
         self.allow_older.update(other_modifiers.allow_older)
 
+    def remove(self, packages):
+        """ Remove all modifiers related to `packages`.
+
+        Parameters
+        ----------
+        packages : an iterable of strings
+            The package names that should be completely removed.
+        """
+        disallowed = (type(b""), type(u""))
+        if isinstance(packages, disallowed):
```

Comment:
shouldn't we instead check for iterable of strings ?

Also, iterable is a bit of a misnomer here, as it will break w/ a non sequence. If we want to make it work for iterables, we should first "reify" the iterable before passing it down.